gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -0,0 +1,441 @@
+    faf_expr,
+    get_adj_expr,
+    pop_max_expr,
+    qual_hist_expr,
+    set_female_y_metrics_to_na_expr,
+)
+from gnomad.utils.file_utils import file_exists
+from gnomad.utils.release import make_faf_index_dict, make_freq_index_dict
+from gnomad.utils.slack import slack_notifications
+
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v4.resources.annotations import (  # TODO: Need to generate this resource for raw callstats
+    get_freq,
+)
+from gnomad_qc.v4.resources.basics import (
+    get_checkpoint_path,
+    get_gnomad_v4_vds,
+    qc_temp_prefix,
+)
+from gnomad_qc.v4.resources.meta import meta
+from gnomad_qc.v4.resources.release import release_sites
+from gnomad_qc.v4.resources.variant_qc import (  # TODO: Confirm we want this in frequency calculations
+    SYNDIP,
+)
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s: %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("gnomAD_frequency")
+logger.setLevel(logging.INFO)
+
+
+def annotate_non_ref_het(vds: hl.vds.VariantDataset) -> hl.vds.VariantDataset:
+    """
+    Annotate non-ref heterozygous calls to prevent incorrect call adjustment post-split.
+
+    :param vds: Hail VDS to annotate non-ref het onto variant data.
+    :return: Hail VDS with _non_ref_het annotation.
+    """
+    vds.variant_data = vds.variant_data.annotate_entries(
+        _het_non_ref=vds.variant_data.LGT.is_het_non_ref()
+    )
+    return vds
+
+
+def annotate_high_ab_hets_by_group_membership(
+    mt: hl.MatrixTable,
+    ab_cutoff: hl.float = 0.9,
+) -> hl.MatrixTable:
+    """
+    Annotate high AB hets by group membership.
+
+    :param mt: MatrixTable to annotate high AB hets onto.
+    :param ab_cutoff: Allele balance cutoff for hom alt depletion fix. Defaults to 0.9
+    :return: _description_
+    """
+    logger.info("Annotating number of high AB het sites in each freq group...")
+    mt = mt.annotate_rows(
+        high_ab_hets_by_group_membership=hl.agg.array_agg(
+            lambda i: hl.agg.filter(
+                mt.group_membership[i]
+                & needs_high_ab_het_fix_expr(
+                    mt,
+                    ab_cutoff,
+                ),
+                hl.agg.count(),
+            ),
+            hl.range(hl.len(mt.group_membership)),
+        )
+    )
+    return mt
+
+
+def needs_high_ab_het_fix_expr(
+    mt: hl.MatrixTable,
+    ab_cutoff: hl.float = 0.9,
+) -> hl.MatrixTable:
+    """
+    Annotate the MatrixTable rows with the count of high AB het genotypes per frequency sample grouping.
+
+    This arrary allows the AC to be corrected with the homalt depletion fix.
+
+    :param mt: Hail MatrixTable to annotate.
+    :param ab_cutoff: Alelle balance threshold at which the GT changes to homalt. Default is 0.9.
+    :return mt:
+    """
+    return (
+        (mt.AD[1] / mt.DP > ab_cutoff)
+        & mt.adj
+        & mt.GT.is_het_ref()
+        & mt.meta.project_meta.fixed_homalt_model
```

Comment:
I think this might need to be 
```suggestion
        & ~mt.meta.project_meta.fixed_homalt_model
```

True means it has a fixed homalt model in the GATK version used

Reply:
That does make sense!